### ISSUE TITLE
IEDriverServer getting crashed after 30 mins

### ISSUE THREAD

**Issue by Venky2005** [SeleniumHQ/selenium#6303]
## Meta -
OS:  
Windows 7 64 Bit
Selenium Version:  
3.14.0
Browser:  
Internet Explorer 9

IEDriverServer consistently getting crashed anytime after 30 mins. If we execute below 30 mins, everything goes well. Whenever the IE get crashed, we get dialog "Command line server for the IE driver has stopped working". I have attached the log below. Please advice. 

Using FindElements atom to locate element having css = #menuTable div[id*='ctl00_mnuMainn3Items'] a[class*='ctl00_mnuMain_1']
T 2018-08-17 00:31:05:140 c:\projects\webdriver\cpp\iedriver\browser.cpp(138) Entering Browser::GetDocument
I 2018-08-17 00:31:05:140 c:\projects\webdriver\cpp\iedriver\browser.cpp(142) No child frame focus. Focus is on top-level frame
T 2018-08-17 00:31:05:154 c:\projects\webdriver\cpp\iedriver\browser.cpp(627) Entering Browser::GetDocumentFromWindow
T 2018-08-17 00:31:05:159 c:\projects\webdriver\cpp\iedriver\script.cpp(60) Entering Script::Initialize
T 2018-08-17 00:31:05:159 c:\projects\webdriver\cpp\iedriver\script.cpp(81) Entering Script::AddArgument(std::wstring)
T 2018-08-17 00:31:05:159 c:\projects\webdriver\cpp\iedriver\script.cpp(116) Entering Script::AddArgument(VARIANT)
T 2018-08-17 00:31:05:159 c:\projects\webdriver\cpp\iedriver\script.cpp(81) Entering Script::AddArgument(std::wstring)
T 2018-08-17 00:31:05:159 c:\projects\webdriver\cpp\iedriver\script.cpp(116) Entering Script::AddArgument(VARIANT)
T 2018-08-17 00:31:05:159 c:\projects\webdriver\cpp\iedriver\script.cpp(180) Entering Script::Execute
T 2018-08-17 00:31:05:159 c:\projects\webdriver\cpp\iedriver\script.cpp(561) Entering Script::CreateAnonymousFunction
W 2018-08-17 00:31:05:168 c:\projects\webdriver\cpp\iedriver\script.cpp(578) -2147024882 [Not enough storage is available to complete this operation.]: Unable to execute code, call to IHTMLWindow2::execScript failed
W 2018-08-17 00:31:05:169 c:\projects\webdriver\cpp\iedriver\script.cpp(191) Cannot create anonymous function
W 2018-08-17 00:31:05:169 c:\projects\webdriver\cpp\iedriver\elementfinder.cpp(164) A JavaScript error was encountered executing the findElements atom.
T 2018-08-17 00:31:05:420 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(1047) Entering IECommandExecutor::LocateElements
T 2018-08-17 00:31:05:420 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(1008) Entering IECommandExecutor::GetElementFindMethod
T 2018-08-17 00:31:05:420 c:\projects\webdriver\cpp\iedriver\elementfinder.cpp(111) Entering ElementFinder::FindElements
T 2018-08-17 00:31:05:420 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(890) Entering IECommandExecutor::GetCurrentBrowser
T 2018-08-17 00:31:05:420 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(896) Entering IECommandExecutor::GetManagedBrowser
T 2018-08-17 00:31:05:420 c:\projects\webdriver\cpp\iedriver\elementfinder.cpp(310) Entering ElementFinder::HasNativeCssSelectorEngine
T 2018-08-17 00:31:05:420 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(890) Entering IECommandExecutor::GetCurrentBrowser
T 2018-08-17 00:31:05:420 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(896) Entering IECommandExecutor::GetManagedBrowser
T 2018-08-17 00:31:05:420 c:\projects\webdriver\cpp\iedriver\browser.cpp(138) Entering Browser::GetDocument
I 2018-08-17 00:31:05:420 c:\projects\webdriver\cpp\iedriver\browser.cpp(142) No child frame focus. Focus is on top-level frame
T 2018-08-17 00:31:05:426 c:\projects\webdriver\cpp\iedriver\browser.cpp(627) Entering Browser::GetDocumentFromWindow
T 2018-08-17 00:31:05:429 c:\projects\webdriver\cpp\iedriver\script.cpp(60) Entering Script::Initialize
T 2018-08-17 00:31:05:429 c:\projects\webdriver\cpp\iedriver\script.cpp(180) Entering Script::Execute
T 2018-08-17 00:31:05:429 c:\projects\webdriver\cpp\iedriver\script.cpp(561) Entering Script::CreateAnonymousFunction
W 2018-08-17 00:31:05:448 c:\projects\webdriver\cpp\iedriver\script.cpp(578) -2147352319 [Unknown error 0x80020101]: Unable to execute code, call to IHTMLWindow2::execScript failed
W 2018-08-17 00:31:05:449 c:\projects\webdriver\cpp\iedriver\script.cpp(191) Cannot create anonymous function
D 2018-08-17 00:31:05:449 c:\projects\webdriver\cpp\iedriver\elementfinder.cpp(118) Element location strategy is CSS selectors, but document does not support CSS selectors. Falling back to using the Sizzle JavaScript CSS selector engine.
T 2018-08-17 00:31:05:449 c:\projects\webdriver\cpp\iedriver\elementfinder.cpp(226) Entering ElementFinder::FindElementsUsingSizzle
T 2018-08-17 00:31:05:449 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(890) Entering IECommandExecutor::GetCurrentBrowser
T 2018-08-17 00:31:05:449 c:\projects\webdriver\cpp\iedriver\iecommandexecutor.cpp(896) Entering IECommandExecutor::GetManagedBrowser
T 2018-08-17 00:31:05:449 c:\projects\webdriver\cpp\iedriver\browser.cpp(138) Entering Browser::GetDocument
I 2018-08-17 00:31:05:449 c:\projects\webdriver\cpp\iedriver\browser.cpp(142) No child frame focus. Focus is on top-level frame
T 2018-08-17 00:31:05:456 c:\projects\webdriver\cpp\iedriver\browser.cpp(627) Entering Browser::GetDocumentFromWindow
T 2018-08-17 00:31:05:460 c:\projects\webdriver\cpp\iedriver\script.cpp(60) Entering Script::Initialize
T 2018-08-17 00:31:05:460 c:\projects\webdriver\cpp\iedriver\script.cpp(81) Entering Script::AddArgument(std::wstring)
T 2018-08-17 00:31:05:460 c:\projects\webdriver\cpp\iedriver\script.cpp(116) Entering Script::AddArgument(VARIANT)
T 2018-08-17 00:31:05:460 c:\projects\webdriver\cpp\iedriver\script.cpp(180) Entering Script::Execute
T 2018-08-17 00:31:05:460 c:\projects\webdriver\cpp\iedriver\script.cpp(561) Entering Script::CreateAnonymousFunction
W 2018-08-17 00:31:34:291 c:\projects\webdriver\cpp\iedriver\script.cpp(578) -2147352567 [Exception occurred.]: Unable to execute code, call to IHTMLWindow2::execScript failed
W 2018-08-17 00:31:34:295 c:\projects\webdriver\cpp\iedriver\script.cpp(191) Cannot create anonymous function
W 2018-08-17 00:31:34:295 c:\projects\webdriver\cpp\iedriver\elementfinder.cpp(303) Execution returned error

Browser Version:  
9.0.8112.16421

## Expected Behavior -
The script should not break due to IEDriverServer crash for long run executions.

## Actual Behavior -
The script break due to IEDriverServer crash for long run executions.

**Comment by GitSage** [comment]
I have an open issue (#5778) about this. I was never able to isolate the exact cause of this problem but I was able to get IE (or IEDriverServer) to consistently fail within 2 minutes by repeatedly loading a page and entering text into a text field. 

I wasn't able to get a single reliable error message though. Sometimes it would look like IE failed, sometimes the command line, sometimes it was unclear what failed.

I had luck with the issue by going back to IEDriverServer v3.8.0. 

Another idea for you to try, assuming your 30 minute IE run isn't all a single test, is to enable `restartBrowserBetweenTests` https://github.com/angular/protractor/blob/7b0808396458fbc2bd46c7e929f4effecb2a3f1e/lib/config.ts#L546

**Comment by Venky2005** [comment]
@GitSage - It is single test running more than 30 minutes based on data. We built our selenium framework with BDD (Specflow). In our case, we get consistent command line error message for the tests runs more than 30 minutes. Do you see any suspicious errors in IE driver log attached in my initial issue thread?

**Comment by mjdavies** [comment]
I'm seeing the same thing, did anyone get a resolution to this?

**Comment by GitSage** [comment]
I ended up convincing the boss that automated testing against Internet Explorer was a bad idea. There are so many bugs, it's so slow, it's so expensive to run many IE instances at once, etc, etc, etc. 

So yeah, no progress.

**Comment by mjdavies** [comment]
@GitSage thanks for replying.  I wish I could talk my boss around, but the user base default browser is IE, arghhhhhhhh!!!

I secretly suspect they all use chrome anyway, but what can you do?

**Comment by GitSage** [comment]
Our clients are all on IE too. I basically sat down and said "look boss, it's just not going to work. It's never going to run reliably, we're ALWAYS going to have to test it manually, so let's just run against Chrome to try to catch most issues, and have QA do their manual runs against IE occasionally." 

Before that I spent about two months, full-time, trying to get IE to run stably and with arbitrary synchroneity and never got it past about an 80% success rate on short tests, with longer tests usually failing.

My boss is a cool guy though. Listens to his employees. He trusts me, plus I had backup from multiple other devs, so that's how I was able to convince him.
